x-pack/plugins/task_manager/server/plugin.ts

```diff
@@ -229,7 +231,18 @@ export class TaskManagerPlugin
     executionContext,
     docLinks,
   }: CoreStart): TaskManagerStartContract {
-    const savedObjectsRepository = savedObjects.createInternalRepository(['task']);
+    const savedObjectsRepository = savedObjects.createInternalRepository([
+      TASK_SO_NAME,
+      BACKGROUND_TASK_NODE_SO_NAME,
+    ]);
+
+    this.kibanaDiscoveryService = new KibanaDiscoveryService({
+      savedObjectsRepository,
+      logger: this.logger,
+      currentNode: this.taskManagerId!,
+    });
+    this.kibanaDiscoveryService.startDiscovery().catch(() => {});
+    this.kibanaDiscoveryService.startCleanup().catch(() => {});
```

Comment:
I realized we should only start the discover service within the `if (shouldRunBackgroundTasks) { ... }` otherwise we'll discover UI only nodes too.

Reply:
done